storage/src/main/java/com/tencent/rss/storage/factory/ShuffleHandlerFactory.java

```diff
@@ -56,7 +58,28 @@ public static synchronized ShuffleHandlerFactory getInstance() {
     return INSTANCE;
   }
 
+  public ClientReadHandler createComposedReadHandler(
+      String[] types, CreateShuffleReadHandlerRequest request) {
+    int size = types.length;
+    ClientReadHandler[] handlers = new ClientReadHandler[size];
+    for (int i = 0; i < size; i++) {
+      request.setStorageType(types[i]);
+      handlers[i] = createSingleReadHandler(request);
+    }
+    return new ComposedClientReadHandler(handlers);
+  }
+
   public ClientReadHandler createShuffleReadHandler(CreateShuffleReadHandlerRequest request) {
+    String storageType = request.getStorageType();
+    String[] types = storageType.split(",");
+    if (types.length > 1) {
+      return createComposedReadHandler(types, request);
+    } else {
+      return createSingleReadHandler(request);
```

Comment:
Seemly we don't need SingleReadHandler

Reply:
There has no SingleReadHandler, this method will be refactor